ASIM/dev/ASimTester/filteringTest/ASimFilteringTest.py

```diff
@@ -218,6 +273,105 @@ def scalar_test_check_fictive_value(self, param_name, query_definition, column_n
             self.assertEqual(0, len(no_results_response.tables[0].rows), f"Parameter: {param_name} - Returned results for non existing filter value. Filtered by value: {DUMMY_VALUE}")
 
         
+    # Returning an array of at most two values from the query response, each string in the returned array is not contained in the other
+    def get_values_for_dynamic_tests(self, query_response):
+        if len(query_response.tables[0].rows) == 1:
+            if query_response.tables[0].rows[0][0] == "":
+                return []
+            return [query_response.tables[0].rows[0][0]]
+        if len(query_response.tables[0].rows) == 2:
+            # Returning only one value in the array if one string is contained in the second
+            if query_response.tables[0].rows[0][0] in query_response.tables[0].rows[0][1]:
+                return [query_response.tables[0].rows[0][1]]
+            if query_response.tables[0].rows[0][1] in query_response.tables[0].rows[0][0]:
+                return [query_response.tables[0].rows[0][0]]
+            return [query_response.tables[0].rows[0][0], query_response.tables[0].rows[1][0]]
+        values = []
+        # Searching values in the query response which are not contained in at least one other value
+        for row in query_response.tables[0].rows:
+            value = row[0]
+            if value == "":
+                continue
+            for row2 in query_response.tables[0].rows:
+                value2 = row2[0]
+                if value not in value2:
+                    values.append(value)
+                    break
+            if len(values) == 2:
+                break
+        return values
```

Comment:
let's talk about this, it is not understood why you are dealing with the case where the length is 1, 2. 
why the general approach is not good enough

Reply:
For the case of length 2 you are correct. I will remove it.
For the case of length 1 lets assume we have a response with only one value. The if condition in line 297 (if value not in value2) will not be true as both "value" and "value2" will have the value of the only element in the response.
As a result, this one value won't be appended to the returned array even though it should be appended.
By the way, following our teams meeting, this is the same reason why the case of length 1 is checked in both get_prefix and get_postfix.

Reply:
Let's talk

Reply:
def get_values_for_dynamic_tests(self, rows):
	if len(rows) == 1:
		return [] if rows[0][0] == "" else return [rows[0][0]]
		
	values = []
	# Searching values in rows which are not contained in at least one other value
	for row in rows:
		value = row[0]
		# if the value in an empty string we want to skip it
		if value == "":
			continue
			
		# if we already found 2 values that sutisfy the condifitons we can return them 
		if len(values) == 2:
			break
			
		for rowToCompareWith in rows:
			valueToCompareWith = rowToCompareWith[0]
			if value not in valueToCompareWith:
				values.append(value)
				# if we found one value that sutisfy the condition, we can continue to the next one
				break
	return values

Reply:
Changed the function after talking face to face.